lib/stm32/l1/flash.c

```diff
@@ -230,5 +230,50 @@ void eeprom_program_words(uint32_t address, uint32_t *data, int length_in_words)
 	flash_lock_pecr();
 }
 
+/** @brief Erase a page (256 Byte) of program memory (flash)
+ *
+ * @param[in] page_address must point to flash space, no checking!
+ */
+void flash_erase_page(uint32_t address)
+{
+	while (FLASH_SR & FLASH_SR_BSY);
+	FLASH_PECR |= (FLASH_PECR_ERASE | FLASH_PECR_PROG);
+	*(uint32_t*)address = 0;
```

Comment:
Right it must be the beginning of the page. About the pointer, I was just following existing code in flash.c of other prcessors (f1/f0...) and in the same file `void eeprom_program_word(uint32_t address, uint32_t data)`